File: frontend/src/api/cloud.ts
import client from './client'

export interface CloudDestination {
  id: number
  name: string
  provider: string
  endpoint: string
  region: string | null
  bucket: string
  prefix: string
  secure: boolean
  enabled: boolean
  created_at: string
}
export interface SyncTarget {
  id: number
  connection_id: number
  cloud_destination_id: number
  enabled: boolean
}
export interface BackupFile { id: number; connection_id: number; status: string; file_path: string | null }

export const listDestinations = () => client.get<CloudDestination[]>('/cloud-destinations')
export const createDestination = (data: Record<string, unknown>) => client.post<CloudDestination>('/cloud-destinations', data)
export const deleteDestination = (id: number) => client.delete(`/cloud-destinations/${id}`)
export const testDestination = (id: number) => client.post(`/cloud-destinations/${id}/test`)
export const listTargets = () => client.get<SyncTarget[]>('/sync-targets')
export const createTarget = (data: Record<string, unknown>) => client.post<SyncTarget>('/sync-targets', data)
export const deleteTarget = (id: number) => client.delete(`/sync-targets/${id}`)
export const syncRun = (backup_record_id: number) => client.post('/sync/run', { backup_record_id })

File: app/schemas/cloud.py
from __future__ import annotations
from datetime import datetime
from pydantic import BaseModel


class CloudDestinationCreate(BaseModel):
    name: str
    provider: str = "s3"
    endpoint: str
    region: str | None = None
    bucket: str
    access_key: str
    secret: str
    prefix: str = ""
    secure: bool = True
    enabled: bool = True


class CloudDestinationOut(BaseModel):
    id: int
    name: str
    provider: str
    endpoint: str
    region: str | None
    bucket: str
    prefix: str
    secure: bool
    enabled: bool
    created_at: datetime
    # 注意:不含 access_key / secret —— 永不回传

    model_config = {"from_attributes": True}


class SyncTargetCreate(BaseModel):
    connection_id: int
    cloud_destination_id: int
    enabled: bool = True


class SyncTargetOut(BaseModel):
    id: int
    connection_id: int
    cloud_destination_id: int
    enabled: bool

    model_config = {"from_attributes": True}


class SyncRunRequest(BaseModel):
    backup_record_id: int
